subprojects/core-api/src/main/java/org/gradle/api/internal/cache/CacheResourceConfigurationInternal.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.cache;
+
+import org.gradle.api.cache.CacheResourceConfiguration;
+
+import java.util.function.Supplier;
+
+public interface CacheResourceConfigurationInternal extends CacheResourceConfiguration {
+    Supplier<Long> getRemoveUnusedEntriesOlderThanAsSupplier();
```

Comment:
I wondered if it might be an issue with not having the type available.  No big deal here.

Perhaps extracting the `org.gradle.api.provider` package into a new subproject so that it is usable from everywhere, even projects that are depped by `cope-api`, might be a good idea (outside of this PR)?  Seems like those types shouldn't have any dependencies, and should be usable everywhere, for situations exactly like this one.